server/broadcast_test.go

```diff
@@ -1114,6 +1127,160 @@ func TestProcessSegment_MaxAttempts(t *testing.T) {
 	assert.Len(bsm.sessMap, 0)
 }
 
+type queueEvent struct {
+	key  string
+	data interface{}
+}
+
+type ProducerChan struct {
```

Comment:
Unless we want to make this part of the `server` package API, I suggest we make this private i.e. `producerChan`. We can always export it later if the need arises later, but I personally like the default of keeping it private unless we know we want to expose it as a part of the public API.

Reply:
Oh true, not sure why I made (and kept) it private